web/src/utils/metrics.ts

```diff
@@ -0,0 +1,204 @@
+  range: number | TimeRange,
+  scope: MetricScope
+): TopologyMetrics[] => {
+  const metrics = raw.map(r => parseMetric(r, range, scope));
+
+  // Disambiguate display names with kind when necessary
+  if (scope === 'owner' || scope === 'resource') {
+    // Define some helpers
+    const getKind = scope === 'owner' ? (p: TopologyMetricPeer) => p.ownerType : (p: TopologyMetricPeer) => p.type;
+    const addKind = (p: TopologyMetricPeer) => {
+      if (p.displayName) {
+        let existing = nameKinds.get(p.displayName);
+        if (!existing) {
+          existing = new Set();
+          nameKinds.set(p.displayName, existing);
+        }
+        const k = getKind(p);
+        if (k) {
+          existing.add(k);
+        }
+      }
+    };
+    const disambiguate = (p: TopologyMetricPeer) => {
+      if (p.displayName) {
+        const kinds = nameKinds.get(p.displayName);
+        if (kinds && kinds.size > 1) {
+          let k = getKind(p);
+          if (k) {
+            k = shortKindMap[k] || k.toLowerCase();
+            p.displayName = `${p.displayName} (${k})`;
+          }
+        }
+      }
+    };
+
+    // First pass: extract all names+kind couples
+    const nameKinds = new Map<string, Set<string>>();
+    metrics.forEach(m => {
+      addKind(m.source);
+      addKind(m.destination);
+    });
+
+    // Second pass: update names if necessary
+    metrics.forEach(m => {
+      disambiguate(m.source);
+      disambiguate(m.destination);
+    });
+  }
+  return metrics;
+};
+
+const parseMetric = (raw: RawTopologyMetrics, range: number | TimeRange, scope: MetricScope): TopologyMetrics => {
+  const stats = computeStats(raw.values, range);
+  const source: TopologyMetricPeer = {
+    addr: raw.metric.SrcAddr,
+    name: raw.metric.SrcK8S_Name,
+    namespace: raw.metric.SrcK8S_Namespace,
+    ownerName: raw.metric.SrcK8S_OwnerName,
+    ownerType: raw.metric.SrcK8S_OwnerType,
+    type: raw.metric.SrcK8S_Type,
+    hostName: raw.metric.SrcK8S_HostName
+  };
+  const destination: TopologyMetricPeer = {
+    addr: raw.metric.DstAddr,
+    name: raw.metric.DstK8S_Name,
+    namespace: raw.metric.DstK8S_Namespace,
+    ownerName: raw.metric.DstK8S_OwnerName,
+    ownerType: raw.metric.DstK8S_OwnerType,
+    type: raw.metric.DstK8S_Type,
+    hostName: raw.metric.DstK8S_HostName
+  };
+  source.displayName = buildPeerDisplayName(source, scope);
+  destination.displayName = buildPeerDisplayName(destination, scope);
+  return {
+    source: source,
+    destination: destination,
+    values: raw.values,
+    stats: stats
+  };
+};
+
+/**
+ * computeStats computes avg, max and total. Input metric is always the bytes rate (Bps).
+ */
+export const computeStats = (values: (string | number)[][], range: number | TimeRange): MetricStats => {
+  const filtered = values.map(dp => Number(dp[1])).filter(v => !_.isNaN(v));
+  if (filtered.length === 0) {
+    return { latest: 0, avg: 0, max: 0, total: 0 };
+  }
+
+  // TODO: fill missing dp with 0
```

Comment:
should we do that in a followup ?

Reply:
I would see something like this:

```typescript
  //fill missing dp with 0
  const startRange = endRange - rangeInSeconds;
  let current = startRange;
  while (current <= endRange) {
    current = current + info.stepSeconds;
    if (!values.find(rv => rv[0] as number > current - tolerance && rv[0] as number < current + tolerance)) {
      values.push([current, '0']);
    }
  }
  values = values.sort((a, b) => (a[0] as number) - (b[0] as number));
```

Reply:
Planned already :) https://issues.redhat.com/browse/NETOBSERV-611